### ISSUE TITLE
Loop [WS-Client] connection lost when connectd to host socket.io

### ISSUE THREAD

**Issue by tinamore** [Links2004/arduinoWebSockets#167]
Hi,
I have simple create nodejs nodejs
```
var server = require('http').createServer(});
var io = require('socket.io')(server);
io.on('connection', function(socket) {  
  console.log('-------');
});
server.listen(3000);
console.log('Server run on port 3000');
```
when run WebSocketClientSocketIO.ino on esp8266 then show error log
```
.[WS-Client] connect ws...
[WS-Client] connected to reserve-hk.skyhotel.vn:3000.
[WS-Client][sendHeader] sending header...
[WS-Client][sendHeader] sending header... Done (53087us).
[WS-Client][handleHeader] RX: HTTP/1.1 200 OK
[WS-Client][handleHeader] RX: Content-Type: application/octet-stream
[WS-Client][handleHeader] RX: Content-Length: 101
[WS-Client][handleHeader] RX: Access-Control-Allow-Credentials: true
[WS-Client][handleHeader] RX: Access-Control-Allow-Origin: file://
[WS-Client][handleHeader] RX: Set-Cookie: io=8a8OrVelIerYx6o5AAAB; Path=/; HttpOnly
[WS-Client][handleHeader] RX: Date: Sun, 29 Jan 2017 10:10:22 GMT
[WS-Client][handleHeader] RX: Connection: keep-alive
[WS-Client][handleHeader] Header read fin.
[WS-Client][handleHeader] Client settings:
[WS-Client][handleHeader]  - cURL: /socket.io/?EIO=3
[WS-Client][handleHeader]  - cKey: eon1q5IfCZC2GHV/eDyV4A==
[WS-Client][handleHeader] Server header:
[WS-Client][handleHeader]  - cCode: 200
[WS-Client][handleHeader]  - cIsUpgrade: 0
[WS-Client][handleHeader]  - cIsWebsocket: 1
[WS-Client][handleHeader]  - cAccept: 
[WS-Client][handleHeader]  - cProtocol: arduino
[WS-Client][handleHeader]  - cExtensions: 
[WS-Client][handleHeader]  - cVersion: 0
[WS-Client][handleHeader]  - cSessionId: 8a8OrVelIerYx6o5AAAB; Path=/; HttpOnly
[WS-Client][sendHeader] sending header...
[WS-Client][sendHeader] sending header... Done (54204us).
[WS-Client][handleHeader] RX: 
[WS-Client] connection lost.
[WS-Client] client disconnected.
[WSc] Disconnected!
[WS-Client] connect ws...
[WS-Client] connected to reserve-hk.skyhotel.vn:3000.
[WS-Client][sendHeader] sending header...
[WS-Client][sendHeader] sending header... Done (53676us).
[WS-Client] connection lost.
[WS-Client] client disconnected.
[WSc] Disconnected!
[WS-Client] connect ws...
[WS-Client] connected to reserve-hk.skyhotel.vn:3000.
[WS-Client][sendHeader] sending header...
[WS-Client][sendHeader] sending header... Done (148963us).
[WS-Client] connection lost.
[WS-Client] client disconnected.
```
Same problem in https://github.com/Links2004/arduinoWebSockets/issues/140

how fix error ?

**Comment by kkojabee** [comment]
While working on wemos d1 r1 with iisnode, I found that sessionId parsing is incorrect on handshaking.
From your log, cSessionId should be "8a8OrVelIerYx6o5AAAB" not "8a8OrVelIerYx6o5AAAB; Path=/; HttpOnly"

I changed WebSocketsClient.cpp file on this library for solving this issue.
`client->cSessionId = headerValue.substring(headerValue.indexOf('=') + 1);`
to
`if (headerValue.indexOf("HttpOnly") > -1) {
	client->cSessionId = headerValue.substring(headerValue.indexOf('=') + 1, headerValue.indexOf(";"));
}
else {
	client->cSessionId = headerValue.substring(headerValue.indexOf('=') + 1);
}`
I wish this would help you.
